src/extension.ts

```diff
@@ -364,7 +364,9 @@ class ExtensionManager implements vscode.Disposable {
     const cmt = info.cmakeTools;
 
     // Silent re-scan when detecting a breaking change in the kits definition.
-    const silentScanForKitsNeeded: boolean = await scanForKitsIfNeeded(cmt.extensionContext);
+    // Do this only for the first folder, to avoid multiple rescans taking place in a multi-root workspace.
+    const silentScanForKitsNeeded: boolean = vscode.workspace.workspaceFolders![0] === cmt.folder &&
```

Comment:
You should always check whether the `workspaceFolders` is set or not before accessing the folders.  There are some cases where it can be undefined.